### PR TITLE
[SITE] Introduce LinkButton to avoid duplicate focus states

### PR DESCRIPTION
**Pull request by kachkaev**
When working on the element focus styles, I noticed one interesting UI artifact: some buttons had two tab stops:

Screenshot 2022-01-26 at 18 37 25

Screenshot 2022-01-26 at 18 37 59

This was caused by `<button>` elements sitting inside `<a>` elements.

---

This PR introduces `LinkButton`, which replaces `<Link><Button /></Link>` and fixes tab stops:

Screenshot 2022-01-26 at 19 20 09

Red outlines were created with `*:focus { outline: 1px dotted red !important; }`. I did not commit this CSS, so the buttons should look and behave as before. Visible changes are mentioned in PR comments.

### CODE REVIEW

File: site/src/components/pages/home/Section2/index.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { Step1, Step2, Step3, Step4 } from "./Steps";
 const CONTENT = [
   {
     id: 1,
-    title: "What do you mean by 'blocks'?",
+    title: "What do you mean by ‘blocks’?",
```
Comment (kachkaev) [inline site/src/components/pages/home/Section2/index.tsx:15]: Bonus: [smart quotes](https://smartquotesforsmartpeople.com/)

File: site/src/components/pages/home/WhyBlockProtocol1Section.tsx [hunk 2]
```diff
@@ -41,6 +41,7 @@ const TodoListBlock = () => {
               <Checkbox
                 sx={{ padding: 0, pr: 1.5 }}
                 checked={[2, 3].includes(index)}
+                tabIndex={-1}
```
Comment (kachkaev) [inline site/src/components/pages/home/WhyBlockProtocol1Section.tsx:44]: Bonus: remove tab stops from checkboxes in homepage examples

File: site/src/components/Footer.tsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ const FooterNavigationLinks = FOOTER_NAVIGATION_LINKS.map(({ href, name }) => (
     component="p"
     variant="bpSmallCopy"
     sx={{
+      color: ({ palette }) => palette.gray[40],
```
Comment (kachkaev) [inline site/src/components/Footer.tsx:45]: Bonus: fix link visibility in the footer

Screenshot 2022-01-26 at 19 27 54

Screenshot 2022-01-26 at 19 28 00

File: site/src/components/Link.tsx [hunk 4]
```diff
@@ -67,12 +74,16 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
...
-      (href === "/discord" ||
-        !/^(mailto:|#|\/|https:\/\/blockprotocol\.org)/.test(href));
+    if (process.env.NODE_ENV !== "production") {
+      const children = other.children;
+      if (React.isValidElement(children) && children.type === Button) {
+        throw new Error(
+          "Please use <LinkButton /> instead of <Link><Button /></Link>",
+        );
+      }
+    }
```
Comment (kachkaev) [inline site/src/components/Link.tsx:84]: Usage of `<Link><Button /></Link>` will now crash in dev.

**Review by CiaranMn — APPROVED** [review]
Works well, looks good - nice bonuses 😄

**Pull request merged**